sui_programmability/examples/nft_standard/README.md

```diff
@@ -0,0 +1,114 @@
+# NFT Standards, Philosophy, and Requirements
+
+# Basics
+
+Sui’s object-centric data model gives Sui objects many of the features of a basic ERC721-style NFT: an owner, the ability to be transferred, a globally unique identifier.
+
+### Individual NFT’s
+
+An NFT is a Sui object (i.e., a struct value whose declared type has the `key` ability, which in turn means the struct value has a field named `id` holding a globally unique ID). All NFT’s are Sui objects, but not every Sui object is an NFT.
+
+### Collections
+
+An NFT collection is a Move struct type with the `key` ability. Every NFT collection has a distinct Move struct type, but not every Move struct type corresponds to an NFT collection.
```

Comment:
- We like the idea that NFTs of a given NFT Collection have their own type `T` which is expressed in their Collection type `Collection<T>` , `T` being a witness type representing the NFT type
- Our understanding is that this is useful for human readability as well as to facilitate clients distinguish which collection a given NFT belongs to. Do we miss any other benefits?
- The result of this is that each NFT creator will have to deploy its own module. This is fine and can be abstracted by an SDK, nevertheless these modules should be as light as possible (if we assume 1KB per deployed module 50,000 collections would roughly equate to 50 MB, that’s fine)
- We believe this module deployed by the NFT creators should serve solely as a type exporter and should not contain any custom logic. Instead, the custom logic would be offloaded to another layer of modules that do not need to be deployed every time there is a new collection

Looking into the example implementation we see the following types:

Collection: `Collection<SuimarineNft>`
NFT: `SuimarineNft`

Where `SuimarineNft` would be the object type NFTs of a given NFT Collection. We had in mind the following:

Collection: `Collection<SuimarineNft>`
NFT: `Nft<SuimarineNft>`

This would allow us to leverage the base layer module and its type `Nft` as a type unifier that can be used across the full spectrum of liquidity layer modules. Moreover, we are currently in the process of merging a newly improved design of our `nft-protocol` and believe this Single Witness Pattern could be implemented on top of its base contracts. In this new design there is a base module NFT and we implement custom NFT types on top of it (i.e. Unique NFTs, Collectibles, Composable NFTs, Tickets, etc.). We could therefore have the following three layered approach:

`BaseLayerNft` ← `CustomLayerNft` ← `SingleWitness`

If we wanted to build a collection of NFT collectibles we could have:
`Nft` ← `Collectibles` ← `Suimarine`

Or if we wanted it to be a unique nft collection instead:
`Nft` ← `UniqueNft` ← `Suimarine`